phoenix-core/src/main/java/org/apache/phoenix/schema/SchemaExtractionTool.java

```diff
@@ -0,0 +1,473 @@
+            }
+        }
+        return coveredColumnsBuilder.toString();
+    }
+
+    protected String generateIndexDDLString(String baseTableFullName, String indexedColumnString, String coveredColumnString, boolean local, String pSchemaName, String pTableName) {
+        StringBuilder outputBuilder = new StringBuilder(String.format(CREATE_INDEX, local ? "LOCAL " : "", pTableName, baseTableFullName));
+        outputBuilder.append("(");
+        outputBuilder.append(indexedColumnString);
+        outputBuilder.append(")");
+        if(!coveredColumnString.equals("")) {
+            outputBuilder.append(" INCLUDE (");
+            outputBuilder.append(coveredColumnString);
+            outputBuilder.append(")");
+        }
+        return outputBuilder.toString();
+    }
+
+    PTable getPTable(String pTableFullName) throws SQLException {
+        try (Connection conn = getConnection()) {
+            return PhoenixRuntime.getTable(conn, pTableFullName);
+        }
+    }
+
+    protected String extractCreateViewDDL(PTable table) throws SQLException {
+        String pSchemaName = table.getSchemaName().getString();
+        String pTableName = table.getTableName().getString();
+        String baseTableName = table.getParentTableName().getString();
+        String baseTableFullName = SchemaUtil.getQualifiedTableName(pSchemaName, baseTableName);
+        PTable baseTable = getPTable(baseTableFullName);
+        String columnInfoString = getColumnInfoString(table, baseTable);
+
+        String whereClause = table.getViewStatement();
+        if(whereClause != null) {
+            whereClause = whereClause.substring(whereClause.indexOf("WHERE"));
+        }
+        return generateCreateViewDDL(columnInfoString, baseTableFullName, whereClause == null ? "" : " "+whereClause, pSchemaName, pTableName);
+    }
+
+    private String generateCreateViewDDL(String columnInfoString, String baseTableFullName, String whereClause, String pSchemaName, String pTableName) {
+        String viewFullName = SchemaUtil.getQualifiedTableName(pSchemaName, pTableName);
+        StringBuilder outputBuilder = new StringBuilder(String.format(CREATE_VIEW, viewFullName, columnInfoString, baseTableFullName, whereClause));
+        return outputBuilder.toString();
+    }
+
+    public String extractCreateTableDDL(PTable table) throws IOException, SQLException {
+
+        String pSchemaName = table.getSchemaName().getString();
+        String pTableName = table.getTableName().getString();
+
+        ConnectionQueryServices cqsi = getCQSIObject();
+        HTableDescriptor htd = getHTableDescriptor(cqsi, table);
+        HColumnDescriptor hcd = htd.getFamily(SchemaUtil.getEmptyColumnFamily(table));
+
+        populateDefaultProperties(table);
+        setPTableProperties(table);
+        setHTableProperties(htd);
+        setHColumnFamilyProperties(hcd);
+
+        String columnInfoString = getColumnInfoString(table, null);
```

Comment:
might be better if we have 2 different APIs getColumnInfoStringForTable and getColumnInfoStringForView rather than using null